src/main/java/ca/on/oicr/pineryreports/reports/impl/LanesBillingReport.java

```diff
@@ -0,0 +1,464 @@
+    return REPORT_NAME;
+  }
+  
+  @Override
+  public Collection<Option> getOptions() {
+    return Sets.newHashSet(OPT_AFTER, OPT_BEFORE);
+  }
+
+  @Override
+  public void processOptions(CommandLine cmd) throws ParseException {
+    if (cmd.hasOption(OPT_AFTER.getLongOpt())) {
+      String after = cmd.getOptionValue(OPT_AFTER.getLongOpt());
+      if (!after.matches(DATE_REGEX)) {
+        throw new ParseException("After date must be in format yyyy-mm-dd");
+      }
+      this.start = after;
+    }
+    
+    if (cmd.hasOption(OPT_BEFORE.getLongOpt())) {
+      String before = cmd.getOptionValue(OPT_BEFORE.getLongOpt());
+      if (!before.matches(DATE_REGEX)) {
+        throw new ParseException("Before date must be in format yyyy-mm-dd");
+      }
+      this.end = before;
+    }
+  }
+
+  @Override
+  public String getTitle() {
+    return "Lanes Completion (Billing) "
+        + (start == null ? "Any Time" : start)
+        + " - "
+        + (end == null ? "Now" : end);
+  }
+  
+  private static class ReportObject {
+    private final List<DetailedObject> detailedData;
+    private final List<Map<String, Map<String, BigDecimal>>> summaryAsList;
+    
+    public ReportObject(List<DetailedObject> detailed, Map<String, Map<String, BigDecimal>> summary) {
+      detailed.sort(DetailedObject.detailedComparator);
+      this.detailedData = detailed;
+      this.summaryAsList = listifySummary(summary);
+    }
+    
+    public List<DetailedObject> getDetailedData() {
+      return detailedData;
+    }
+    
+    public List<Map<String, Map<String, BigDecimal>>> getSummaryAsList() {
+      return summaryAsList;
+    }
+  }
+  
+  private ReportObject completed;
+  private ReportObject failed;
+
+  @Override
+  protected void collectData(PineryClient pinery) throws HttpResponseException {
+    // samples and instruments and instrument models will come in handy
+    Map<String, SampleDto> samplesById = mapSamplesById(pinery.getSample().all());
+    Map<Integer, InstrumentDto> instrumentsById = pinery.getInstrument().all().stream()
+        .collect(Collectors.toMap(InstrumentDto::getId, dto->dto));
+    Map<Integer, InstrumentModelDto> instrumentModelsById = pinery.getInstrumentModel().all().stream()
+        .collect(Collectors.toMap(InstrumentModelDto::getId, dto->dto));
+    // filter runs within the date range
+    Set<RunDto> newRuns = pinery.getSequencerRun().all().stream().filter(byEndedBetween(start, end)).collect(Collectors.toSet());
+    Set<RunDto> failedRuns = newRuns.stream().filter(run -> "Failed".equals(run.getState())).collect(Collectors.toSet());
+    newRuns.removeAll(failedRuns);
+    
+    completed = getReportAndSummaryData(newRuns, samplesById, instrumentsById, instrumentModelsById);
+    failed = getReportAndSummaryData(failedRuns, samplesById, instrumentsById, instrumentModelsById);
+  }
+  
+  private ReportObject getReportAndSummaryData(Collection<RunDto> runs, Map<String, SampleDto> samplesById,
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 34 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)